From 1ae17c8504a28c2a5f406ba902fa4303c3d436b5 Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Mon, 24 Apr 2023 16:54:54 +0530
Subject: [PATCH] ath12k: Fix packet drop due to invalid HW index

While bringup monitor mode along with the other interfaces, all the packets
are getting dropped in mac80211 because cfg80211_get_hw_idx_by_freq directly
compares the given center frequency with the PHY supported frequency list
instead of comparing the range of frequency.

Fix the cfg80211_get_hw_idx_by_freq logic by comparing the given frequency
with the range of frequency. Since the hardware advertised supported channel
list is directly used for this comparision, aligned the 6GHz channel in the
increasing order.

Fixes: I86af243e7 (cfg80211/mac80211: add mlo monitor support)
Change-Id: I6c84ee677b17e6fb0abcfbe8f0d39d9f7b287b3f
Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>

Signed-off-by: Yuvasree Sivasankaran <quic_ysivasan@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -114,6 +114,7 @@ static const struct ieee80211_channel at
 };
 
 static const struct ieee80211_channel ath12k_6ghz_channels[] = {
+	CHAN6G(2, 5935, 0),
 	CHAN6G(1, 5955, 0),
 	CHAN6G(5, 5975, 0),
 	CHAN6G(9, 5995, 0),
@@ -173,9 +174,6 @@ static const struct ieee80211_channel at
 	CHAN6G(225, 7075, 0),
 	CHAN6G(229, 7095, 0),
 	CHAN6G(233, 7115, 0),
-
-	/* new addition in IEEE Std 802.11ax-2021 */
-	CHAN6G(2, 5935, 0),
 };
 
 static struct ieee80211_rate ath12k_legacy_rates[] = {
